rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf

```diff
@@ -558,21 +558,30 @@ SecRule ARGS_NAMES|ARGS|XML:/* "@rx (?i)(?:not\s+between\s+(?:(?:(?:'[^']*')|(?:
 #
 # -=[ SQL Tautologies ]=-
 #
-# Regexp generated from util/regexp-assemble/data/942130.data using Regexp::Assemble.
-# To rebuild the regexp:
-#   cd util/regexp-assemble
-#   ./regexp-assemble.py 942130
+# Boolean-based SQL injection or tautology attack. Boolean values (True or False) are used to carry out
+# this type of SQL injection. The malicious SQL query forces the web application to return a different result de-
+# pending on whether the query returns a TRUE or FALSE result.
+#
+# The original 942130 was split in two rules:
+# - 942130 targets tautologies using equalities (e.g. 1 = 1)
+# - 942131 targets tautologies using inequalities (e.g. 1 != 2)
+#
+# We use captures to check for (in)equality in the regexp. So TX.1 will capture the left hand side (LHS) of the inequality,
+# and TX.2 will capture the right hand side (RHS) of the logical query.
 #
-# Not supported by re2 (backreferences, lookaheads).
+# To regenerate the regexp:
```

Comment:
There's an open PR for cleaning up these comments anyway, so leave it as it is to reduce merge issues.

```suggestion
# Regexp generated from util/regexp-assemble/data/942130.data using Regexp::Assemble.
# To regenerate the regexp:```